### PR TITLE
Feat: Add expect syntax by wrapping assert syntax

### PR DESCRIPTION
**Pull request by Sorikairox**
Contains the following. Sadly couldn't make time to write tests (life happened). 

- [x] [expect(value)](https://jestjs.io/docs/expect#expectvalue)
- [x]  [.not](https://jestjs.io/docs/expect#not)
- [x]  [.resolves](https://jestjs.io/docs/expect#resolves)
- [x]  [.rejects](https://jestjs.io/docs/expect#rejects)
- [x]  [.toBe(value)](https://jestjs.io/docs/expect#tobevalue)
- [ ]  [.toHaveLength(number)](https://jestjs.io/docs/expect#tohavelengthnumber)
- [ ]  [.toHaveProperty(keyPath, value?)](https://jestjs.io/docs/expect#tohavepropertykeypath-value?)
- [x]  [.toBeCloseTo(number, numDigits?)](https://jestjs.io/docs/expect#tobeclosetonumber-numdigits?)
- [x]  [.toBeDefined()](https://jestjs.io/docs/expect#tobedefined)
- [x]  [.toBeFalsy()](https://jestjs.io/docs/expect#tobefalsy)
- [x]  [.toBeGreaterThan(number | bigint)](https://jestjs.io/docs/expect#tobegreaterthannumber--bigint)
- [x]  [.toBeGreaterThanOrEqual(number | bigint)](https://jestjs.io/docs/expect#tobegreaterthanorequalnumber--bigint)
- [x]  [.toBeLessThan(number | bigint)](https://jestjs.io/docs/expect#tobelessthannumber--bigint)
- [x]  [.toBeLessThanOrEqual(number | bigint)](https://jestjs.io/docs/expect#tobelessthanorequalnumber--bigint)
- [x]  [.toBeInstanceOf(Class)](https://jestjs.io/docs/expect#tobeinstanceofclass)
- [x]  [.toBeNull()](https://jestjs.io/docs/expect#tobenull)
- [x]  [.toBeTruthy()](https://jestjs.io/docs/expect#tobetruthy)
- [x]  [.toBeUndefined()](https://jestjs.io/docs/expect#tobeundefined)
- [x]  [.toBeNaN()](https://jestjs.io/docs/expect#tobenan)
- [ ]  [.toContain(item)](https://jestjs.io/docs/expect#tocontainitem)
- [ ]  [.toContainEqual(item)](https://jestjs.io/docs/expect#tocontainequalitem)
- [x]  [.toEqual(value)](https://jestjs.io/docs/expect#toequalvalue)
- [x]  [.toMatch(regexp | string)](https://jestjs.io/docs/expect#tomatchregexp--string)
- [x]  [.toMatchObject(object)](https://jestjs.io/docs/expect#tomatchobjectobject)
- [x]  [.toStrictEqual(value)](https://jestjs.io/docs/expect#tostrictequalvalue)
- [x]  [.toThrow(error?)](https://jestjs.io/docs/expect#tothrowerror?)

related #1779

### CODE REVIEW

**Comment by Sorikairox** [comment]
If anyone has any clue on what the following error mean:

```
Run deno task lint:deprecations
Task lint:deprecations deno run --allow-read --allow-net ./_tools/check_deprecation.ts
Download https://deno.land/x/deno_doc@0.59.0/mod.ts
Download https://deno.land/x/deno_doc@0.59.0/lib/deno_doc.generated.js
Download https://deno.land/x/deno_graph@0.[4](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:5)[5](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:6).0/lib/loader.ts
error: Uncaught (in promise) Error: ../../assert/assertion_error
      const ret = new Error(getStringFromWasm0(arg0, arg1));
                  ^
    at __wbg_new_15d39[6](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:7)6e9981a196 (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc.generated.js:498:19)
    at <anonymous> (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc_bg.wasm:1:2056[7](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:8)[8](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:9)1)
    at <anonymous> (https://deno.land/x/deno_doc@0.5[9](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:10).0/lib/deno_doc_bg.wasm:1:1509332)
    at <anonymous> (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc_bg.wasm:1:1522600)
    at <anonymous> (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc_bg.wasm:1:2008543)
    at __wbg_adapter_46 (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc.generated.js:231:6)
    at real (https://deno.land/x/deno_doc@0.59.0/lib/deno_doc.generated.js:215:14)
    at eventLoopTick (ext:core/01_core.js:183:[11](https://github.com/denoland/deno_std/actions/runs/6106288969/job/16571102052?pr=3627#step:7:12))
Error: Process completed with exit code 1.```

File: testing/_matchers/to_be_falsy.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { MatcherContext, MatchResult } from "../_types.ts";
+import { AssertionError } from "../../assert/assertion_error";
```
Comment (kt3k) [inline testing/_matchers/to_be_falsy.ts:4]:
```suggestion
import { AssertionError } from "../../assert/assertion_error.ts";
```

File: testing/_matchers/to_be_close_to.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { MatcherContext, MatchResult } from "../_types.ts";
+import { assertAlmostEquals } from "../../assert/assert_almost_equals.ts";
+import { AssertionError } from "../../assert/assertion_error";
```
Comment (kt3k) [inline testing/_matchers/to_be_close_to.ts:5]:
```suggestion
import { AssertionError } from "../../assert/assertion_error.ts";
```

File: testing/_matchers/to_be_truthy.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { MatcherContext, MatchResult } from "../_types.ts";
+import { AssertionError } from "../../assert/assertion_error";
```
Comment (kt3k) [inline testing/_matchers/to_be_truthy.ts:4]:
```suggestion
import { AssertionError } from "../../assert/assertion_error.ts";
```

File: assert/assert_instance_of.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@
 import { AssertionError } from "./assertion_error.ts";
 
 // deno-lint-ignore no-explicit-any
-type AnyConstructor = new (...args: any[]) => any;
+export type AnyConstructor = new (...args: any[]) => any;
```
Comment (kt3k) [inline assert/assert_instance_of.ts:5]: Let's not export this, but copy it into expect.ts to avoid increasing the public types.

**Comment by iuioiua** [comment]
Hi @Sorikairox, is this PR ready to merge or did you need a hand with anything?

**Comment by kt3k** [comment]
Thanks for working on this @Sorikairox 

We'll continue the rest of tasks in other branch!

**Comment by kt3k** [comment]
We opened #3814 for continuing the work.

**Pull request closed without merging**